### ISSUE TITLE
unreachable IP prevents timeouts

### ISSUE THREAD

**Issue by vlcinsky** [ufoscout/docker-compose-wait#17]
```
# export WAIT_HOSTS=11.22.33.44:5555
# ./bin/wait && ping google.com
Checking availability of 11.22.33.44:5555
Host 11.22.33.44:5555 not yet available
Host 11.22.33.44:5555 not yet available
...forever
```

The command runs for ever.

I would expect it to timeout and exit after a while.

**Comment by ufoscout** [comment]
Hi @vlcinsky 
I am not able to reproduce it. What version of this library are you using? 
Here what I get:
```
$ WAIT_HOSTS=11.22.33.44:5555 ./wait && ping www.google.com
--------------------------------------------------------
docker-compose-wait - starting with configuration:
 - Hosts to be waiting for: [11.22.33.44:5555]
 - Timeout before failure: 30 seconds 
 - Sleeping time before checking for hosts availability: 0 seconds
 - Sleeping time once all hosts are available: 0 seconds
--------------------------------------------------------
Checking availability of 11.22.33.44:5555
Host 11.22.33.44:5555 not yet available...
Timeout! After 30 seconds some hosts are still not reachable  
```

Could you try with the latest version?

**Comment by vlcinsky** [comment]
The binary I use is definitely different as it produces much less output. I did not manage to determine version I use (there seem to be no `--version` option and `file ./wait` reports `./wait: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), statically linked, BuildID[sha1]=db96d24de4e1db8ecc747117698a609d7e3c4247, stripped` which does not seem to relate to anything like commit hash or version in the git repo).

Anyway, testing with the latest version 2.6.0:
Using script `trywait.sh` as follows:
```
#!/bin/sh
date
WAIT_HOSTS=11.22.33.44:5555 ./wait && ping www.google.com
date

```
and running it
```
$ ./trywait.sh                                                                                                                                                                    
Wed 04 Dec 2019 08:30:27 AM UTC
--------------------------------------------------------
docker-compose-wait - starting with configuration:
 - Hosts to be waiting for: [11.22.33.44:5555]
 - Timeout before failure: 30 seconds
 - Sleeping time before checking for hosts availability: 0 seconds
 - Sleeping time once all hosts are available: 0 seconds
--------------------------------------------------------
Checking availability of 11.22.33.44:5555
Host 11.22.33.44:5555 not yet available...
Timeout! After 30 seconds some hosts are still not reachable
Wed 04 Dec 2019 08:32:37 AM UTC
```
It took 130 seconds before it timed out instead of declared 30 seconds.

**Comment by ufoscout** [comment]
@vlcinsky 
Ok, thanks. At least I know it does not hang forever.
I will investigate asap why it does not respect the timeout.

**Comment by ufoscout** [comment]
Fixed in 2.7.0
